@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -831,159 +876,206 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
+                        const insideExtents = !(
+                            (currX <= 0 && nextX <= 0) ||
+                            (currX >= extents && nextX >= extents) ||
+                            (currY <= 0 && nextY <= 0) ||
+                            (currY >= extents && nextY >= extents)
                         );
-                    }
-                    if (current.contourOutlines !== undefined) {
-                        addPolygonEdges(
-                            edgeIndices,
-                            (isExtruded ? vertexOffset : vertexOffset / 2) + baseVertex,
-                            stride,
-                            current.contour,
-                            current.contourOutlines,
-                            isExtruded,
-                            extrudedPolygonTechnique.footprint,
-                            extrudedPolygonTechnique.maxSlope
+
+                        vertices.push(
+                            insideExtents ? vertexOffset + nextIdx : -1,
+                            boundaryWalls || insideExtents ? vertexOffset + nextIdx : -1
                         );
                     }
                 }
 
                 try {
                     // Triangulate the footprint polyline.
-                    // For sphere projection the stride of Ring.contour and Ring.points can be
-                    // different b/c Ring.contour contains texture coordinates where Ring.points
-                    // does not.
-                    const triangles = earcut(points, holes, stride);
+                    const triangles = earcut(vertices, holes, vertexStride);
+                    const originalVertexCount = vertices.length / vertexStride;
 
+                    // Subdivide for spherical projections if needed.
                     if (isSpherical) {
-                        const geom = new THREE.Geometry();
-
-                        for (let i = 0; i < vertices.length; i += stride) {
-                            geom.vertices.push(
-                                new THREE.Vector3(points[i], points[i + 1], points[i + 2])
-                            );
-                        }
-                        for (let i = 0; i < triangles.length; i += 3) {
-                            geom.faces.push(
-                                new THREE.Face3(triangles[i], triangles[i + 1], triangles[i + 2])
+                        const geom = new THREE.BufferGeometry();
+
+                        const positionArray = [];
+                        const uvArray = [];
+                        const edgeArray = [];
+                        const wallArray = [];
+
+                        // Transform to global webMercator coordinates to be able to reproject to
+                        // sphere.
+                        for (let i = 0; i < vertices.length; i += vertexStride) {
+                            const worldPos = tile2world(
```

Comment:
again, local function

Reply:
Not doing it here, because we don't project to the ``targetProjection`` space here.